### PR TITLE
fix: treat workspace deletion api errors as errors

### PR DESCRIPTION
**Pull request by sebasslash**
## Description

https://github.com/hashicorp/terraform-provider-tfe/pull/675#discussion_r1065176811

### CODE REVIEW

File: tfe/resource_tfe_workspace.go [hunk 1]
```diff
@@ -865,6 +865,7 @@ func errWorkspaceSafeDeleteWithPermission(workspaceID string, err error) error {
 		if strings.HasPrefix(err.Error(), "conflict") {
 			return fmt.Errorf("error deleting workspace %s: %w\nTo delete this workspace without destroying the managed resources, add force_delete = true to the resource config", workspaceID, err)
 		}
+		return err
 	}
 	return nil
```
Comment (brandonc) [inline tfe/resource_tfe_workspace.go:870]: Alternatively 🙃 
```suggestion
	}
	return err
```

**Review by brandonc — APPROVED** [review]
Ew! Easy to miss. Thank you for finding and fixing this!

File: tfe/resource_tfe_workspace.go (hunk 1 above)
Comment (sebasslash) [reply tfe/resource_tfe_workspace.go:870]: :nail_care:

Comment (brandonc) [reply tfe/resource_tfe_workspace.go:870]: Can you also add a CHANGELOG, @sebasslash  ? Thank you

**Comment by skeggse** [comment]
You're welcome!

**Comment by skeggse** [comment]
Is this good to go?

**Comment by sebasslash** [comment]
Wow, this completely flew under my radar! Really sorry!

**Pull request merged**